### PR TITLE
Add a basic DataStructs WorkUnit and associated tests

### PR DESCRIPTION
**Pull request by ericvaandering**

### CODE REVIEW

**Comment by ericvaandering** [comment]
test this please

**Comment by ericvaandering** [comment]
test this please

**Comment by ericvaandering** [comment]
test this please

**Comment by ericvaandering** [comment]
test this please

File: src/python/WMCore/WMBS/Oracle/WorkUnit/Add.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class Add(AddWorkUnitMySQL):
+    """
+    _Add_
+
+    Use Oracle sequence instead of auto_increment and do it in a combined transaction
+    """
+
+    sql = ('INSERT ALL'
```
Comment (ericvaandering) [inline src/python/WMCore/WMBS/Oracle/WorkUnit/Add.py]: @hufnagel I think this is what you were talking about? I looked at your example and you were using SQL to check for existence and return the inserted values as well, but I think this is the heart of "InsertStreamer"

**Comment by ericvaandering** [comment]
@ticoann can you run the unit tests in test/python/WMCore_t/DataStructs_t/WorkUnit_t.py
 and test/python/WMCore_t/WMBS_t/WorkUnit_t.py through Oracle?

**Comment by ticoann** [comment]
@ericvaandering, Eric, I was trying to run the unittest under Oracle. but somehow db connection failed. (Not the code problem but my set up problem). 

```
  File "/data/unittest/unittestdeploy/wmagent/1.1.5.pre3/sw/slc6_amd64_gcc493/external/py2-sqlalchemy/0.9.6-comp4/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 376, in connect
    return self.dbapi.connect(*cargs, **cparams)
DatabaseError: (DatabaseError) ORA-12545: Connect failed because target host or object does not exist
 None None
```
I will try again tomorrow.

**Comment by amaltaro** [comment]
Hey @ericvaandering , I've just ran these tests and the first one is Ok (I ran it 3 times):
```
amaltaro@alan-wmacloud:/data/amaltaro/WMAgent/wmcore_unittest/WMCore/test/python/WMCore_t/DataStructs_t $ nosetests WorkUnit_t.py 
.....
----------------------------------------------------------------------
Ran 5 tests in 0.001s

OK
```

however, the second one is failing all unit tests. You can find the whole log here:
http://amaltaro.web.cern.ch/amaltaro/forEricW/PR_7820/unittest.log

**Comment by ericvaandering** [comment]
Can you try again? The problem is Oracle doesn't let you use AS to alias tables so 

`FROM long_name AS short` should just be `FROM long_name short` which is much less readable. :-(

**Comment by ericvaandering** [comment]
And you need to end INSERT ALL with a select even if you don't care....

**Comment by ericvaandering** [comment]
@ticoann , please merge. @amaltaro says the Oracle tests are fine now too once I fixed two differences between Oracle and MySQL

Unfortunately I can't squash that last commit because I'm using the the one before as the branch point for my next PR.

**Comment by ticoann** [comment]
Thanks Alan for testing. I will merge when Jenkins finish the test.

**Pull request merged**